osu.Game/Screens/OnlinePlay/Components/RoomManager.cs

```diff
@@ -67,11 +57,10 @@ public virtual void CreateRoom(Room room, Action<Room> onSuccess = null, Action<
             {
                 joinedRoom.Value = room;
 
-                update(room, result);
-                addRoom(room);
+                AddOrUpdateRoom(result);
+                room.CopyFrom(result); // Also copy back to the source model, since this is likely to have been stored elsewhere.
 
-                RoomsUpdated?.Invoke();
-                onSuccess?.Invoke(room);
+                onSuccess?.Invoke(result);
```

Comment:
```suggestion
                onSuccess?.Invoke(room);
```

`result` does not have password populated in tests (seemingly purposefully) therefore causing more test failures.

Reply:
Unsure whether this should pass the resultant room or the given room, but I've done it as you suggest for the time being.